ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala

```diff
@@ -104,6 +105,44 @@ class SqlLedgerSpec
       }
     }
 
+    "correctly initialized the participant ID" in {
+      val participantId = makeParticipantId("TheOnlyParticipant")
+      for {
+        _ <- createSqlLedgerWithParticipantId(participantId)
+        metadata <- IndexMetadata.read(postgresDatabase.url)
+      } yield {
+        metadata.participantId shouldEqual participantId
+      }
+    }
+
+    "allow to resume on an existing participant ID" in {
+      val participantId = makeParticipantId("TheParticipant")
+      for {
+        _ <- createSqlLedgerWithParticipantId(participantId)
+        _ <- createSqlLedgerWithParticipantId(participantId)
+        metadata <- IndexMetadata.read(postgresDatabase.url)
+      } yield {
+        metadata.participantId shouldEqual participantId
+      }
+    }
+
+    "refuse to create a new ledger when there is already one with a different participant ID" in {
+      val expectedExisting = makeParticipantId("TheParticipant")
+      val expectedProvided = makeParticipantId("AnotherParticipant")
+      for {
+        _ <- createSqlLedgerWithParticipantId(expectedExisting)
+        throwable <- createSqlLedgerWithParticipantId(expectedProvided).failed
+      } yield {
+        throwable match {
+          case mismatch: MismatchException.ParticipantId =>
+            mismatch.existing shouldEqual expectedExisting
+            mismatch.provided shouldEqual expectedProvided
+          case _ =>
+            fail("Did not get the expected exception type", throwable)
+        }
+      }
+    }
+
```

Comment:
These tests seem to be essentially the same as `JdbcIndexerSpec`. Is there a way to share them?

Reply:
They test the same behavior but for two different integrations that have a significantly different interface. If it makes sense for you, I can come up with a way to factor the behavior out of the integrations and add a unit test for that, but I would prefer keeping both integration tests in.

Reply:
Not if it's too much work, thanks for considering it.